src/Dataset.ts

```diff
@@ -407,28 +447,14 @@ export default class Dataset {
     const quadsString = new n3.Writer().quadsToString(quads); // This always returns nquads or ntriples. See https://github.com/rdfjs/N3.js/issues/253
     const tmpFile = path.resolve(tmpdir(), `triplydb-${md5(quadsString)}.nq`);
     await fs.writeFile(tmpFile, quadsString, "utf-8");
-    return this.importFromFiles(opts || {}, tmpFile);
+    return this.importFromFiles([tmpFile], opts || {});
   }
-  public async importFromUrls(...urls: string[]): Promise<Dataset>;
-  public async importFromUrls(opts: JobDefaultsConfig, ...urls: string[]): Promise<Dataset>;
-  public async importFromUrls(arg1: string | JobDefaultsConfig, ...urls: string[]): Promise<Dataset> {
+  public async importFromUrls(urls: string[], defaultConfig?: JobDefaultsConfig): Promise<Dataset> {
     const dsId = await this.getInfo().then((info) => info.id);
     try {
       this._throwIfJobRunning(dsId);
       datasetsWithOngoingJob[dsId] = true;
-      let baseIRI: string | undefined;
-      let defaultGraphName: string | undefined;
-      let overwriteAll: boolean | undefined;
-      if (typeof arg1 === "string") {
-        // we have just a list of filenames, no defaults config.
-        urls = [arg1].concat(urls);
-      } else {
-        // the first arg is a defaults config.
-        baseIRI = arg1.baseIRI;
-        defaultGraphName = arg1.defaultGraphName;
-        overwriteAll = arg1.overwriteAll;
-      }
-      if (overwriteAll && !(await this._app.isCompatible("2.2.7"))) {
+      if (defaultConfig?.overwriteAll && !(await this._app.isCompatible("2.2.7"))) {
```

Comment:
nitpick: you're using the question mark in quite a few places where you don need one. I.e., `defaultConfig?` can be `defaultConfig`

Reply:
Typescript is complaining at me here, presumably because the `defaultConfig` is optional.